server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldParser.java

```diff
@@ -38,21 +39,35 @@ class FlattenedFieldParser {
     private final int depthLimit;
     private final int ignoreAbove;
     private final String nullValue;
+    private final List<String> dimensions;
 
     FlattenedFieldParser(
         String rootFieldName,
         String keyedFieldName,
         MappedFieldType fieldType,
         int depthLimit,
         int ignoreAbove,
-        String nullValue
+        String nullValue,
+        List<String> dimensions
     ) {
         this.rootFieldName = rootFieldName;
         this.keyedFieldName = keyedFieldName;
         this.fieldType = fieldType;
         this.depthLimit = depthLimit;
         this.ignoreAbove = ignoreAbove;
         this.nullValue = nullValue;
+        this.dimensions = dimensions;
+    }
+
+    FlattenedFieldParser(
+        String rootFieldName,
+        String keyedFieldName,
+        MappedFieldType fieldType,
+        int depthLimit,
+        int ignoreAbove,
+        String nullValue
+    ) {
+        this(rootFieldName, keyedFieldName, fieldType, depthLimit, ignoreAbove, nullValue, Collections.emptyList());
```

Comment:
I'm not usually a huge fan of making extra overrides like this. But it can be super useful to make the PR smaller and easier to review. In that case I'd love a follow up PR to remove the shorter one and update call sites.

Reply:
The PR is longer than it should be mainly because of the changes to the error message in the IllegalArgumentException...sorry for that but I feel like the error must be updated together with the change itself.